src/elysium/sigmaprimitives.cpp

```diff
@@ -101,7 +102,7 @@ bool SigmaPublicKey::operator!=(const SigmaPublicKey& other) const
 
 bool SigmaPublicKey::IsValid() const
 {
-    return commitment.isMember();
+    return commitment.isMember() && !commitment.isInfinity();
```

Comment:
@ultimaweapon to be sure that there is no such commitment, please run a reindex with this code. that will also cover the case when both serial and randomness are 0's.

Reply:
Reindex won't help due to we don't know if there is going to be a new one before everyone is upgraded. So the only way is to activate this new check on a certain block instead.

Reply:
ok, let's do that way

Reply:
This check would effect anon set. I think we should do following.
1) Reject non-member and inf after v1 activated.
2) Reject non-member old coins from anon group but don't refund for non-member mint transaction because that would effect elysium state. If there is some non-member coin in the set users would not be able to spend any coin then they need to reindex to fix anonimity group state.

Reply:
After check whole logic there are no non-member mints are not be added to anon set. Then reindexing is not required.

Reply:
So what we do, we just reject inf commitments after HF block, it is more safe, and exactly there is no need for reindex.

Reply:
The non-member check is already there from the beginning. So the only cases we need to handle is the infinity point checking.

Reply:
@ultimaweapon updated, please check.